Lib/tarfile.py

```diff
@@ -420,6 +420,8 @@ def _init_write_gz(self):
         self.__write(b"\037\213\010\010" + timestamp + b"\002\377")
         if self.name.endswith(".gz"):
             self.name = self.name[:-3]
+        # Remove directory components
+        self.name = os.path.basename(self.name)
```

Comment:
but, if I want to tar a file using the source tree folder, Could I do that?

Reply:
@eamanu Method `_init_write_gz` calls only once when GZIP compressed file opens [(on `__init__` of `_Stream`)](https://github.com/python/cpython/blob/38d3864efe914fda64553e2ec75c9ec15574483f/Lib/tarfile.py#L373).

So you can safely add anything by calling `tar.add('foo/bar')`.

Reply:
@eamanu Also `name` attribute is used only to write header.

Reply:
If that's true, that `name` is only used to write the header, I wonder if the value should be saved on the object at all. It's somewhat out of scope for this change, so I'll submit it as a separate PR.